src/sdam/description/topology/mod.rs

```diff
@@ -275,6 +287,75 @@ impl TopologyDescription {
         }
     }
 
+    /// Updates the topology's logical session timeout value based on the server's value for it.
+    fn update_session_support_status(&mut self, server_description: &ServerDescription) {
+        if !server_description.server_type.is_data_bearing() {
+            return;
+        }
+
+        match server_description.logical_session_timeout().ok().flatten() {
+            Some(timeout) => match self.session_support_status {
+                SessionSupportStatus::Supported {
+                    logical_session_timeout: topology_timeout,
+                } => {
+                    self.session_support_status = SessionSupportStatus::Supported {
+                        logical_session_timeout: std::cmp::min(timeout, topology_timeout),
+                    };
+                }
+                SessionSupportStatus::Undetermined => {
+                    self.session_support_status = SessionSupportStatus::Supported {
+                        logical_session_timeout: timeout,
+                    }
+                }
+                SessionSupportStatus::Unsupported => {
+                    // Check if the timeout is now reported on all servers, and, if so, assign the
+                    // topology's timeout to the minimum.
+                    let min_timeout =
+                        self.servers
+                            .values()
+                            .fold(Some(timeout), |min_so_far, desc| {
```

Comment:
Could this be implemented with `Iterator::min_by`?

Reply:
I can actually just use `min` if I `filter` and `map` first. Updated.

Reply:
Fun fact: there's also [`Iterator::filter_map`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.filter_map), but since it all gets done in a single pass due to laziness, it doesn't make a big difference whether we use it or not.